source/Cosmos.System2/Graphics/DoubleBufferedSVGAIICanvas.cs

```diff
@@ -0,0 +1,336 @@
+//#define COSMOSDEBUG
+using Cosmos.Debug.Kernel;
+using Cosmos.HAL.Drivers.PCI.Video;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+
+namespace Cosmos.System.Graphics
+{
+    /// <summary>
+    /// SVGAIIScreen class. Used to draw ractengales to the screen. See also: <seealso cref="Canvas"/>.
+    /// </summary>
+    public class DoubleBufferedSVGAIICanvas : Canvas
```

Comment:
The idea of a double buffered canva is great. But shouldn't we implement one DoubleBufferedCanvas class that can use the double buffered VBE driver and also the VMWare SVGAII one? @quajak?

Reply:
I think the better design would be that every canvas has a buffered and unbuffered version implemented in the same class and toggled by some bool(at least accesible in the constructor). We can then also have a FullScreenCanvas.GetDoubleBufferedCanvas which returns the double buffered version. I think this is a better design since double buffering implementation depends on the driver but there should be some shared code between the buffered and non buffered implementations.